src/Blazored.Modal/Services/IModalService.cs

```diff
@@ -44,28 +40,28 @@ public interface IModalService
         /// <param name="title">Modal title.</param>
         /// <param name="parameters">Key/Value collection of parameters to pass to component being displayed.</param>
         /// <param name="options">Options to configure the modal.</param>
-        void Show<T>(string title, ModalParameters parameters = null, ModalOptions options = null) where T : ComponentBase;
+        ModalReference Show<T>(string title, ModalParameters parameters = null, ModalOptions options = null) where T : ComponentBase;
 
         /// <summary>
         /// Shows the modal with the specific component type.
         /// </summary>
         /// <param name="contentComponent">Type of component to display.</param>
-        void Show(Type contentComponent);
+        ModalReference Show(Type contentComponent);
 
         /// <summary>
         /// Shows the modal with the component type using the specified title.
         /// </summary>
         /// <param name="contentComponent">Type of component to display.</param>
         /// <param name="title">Modal title.</param>
-        void Show(Type contentComponent, string title);
+        ModalReference Show(Type contentComponent, string title);
 
         /// <summary>
         /// Shows the modal with the component type using the specified title.
         /// </summary>
         /// <param name="title">Modal title.</param>
         /// <param name="contentComponent">Type of component to display.</param>
         /// <param name="options">Options to configure the modal.</param>
-        void Show(Type contentComponent, string title, ModalOptions options);
+        ModalReference Show(Type contentComponent, string title, ModalOptions options);

```

Comment:
Same as above: Maybe make the comments on these also use `<paramref>`, and also mention ModalOptions on the overload with ModalOptions